store/mongo/migration_2_0_0.go

```
@@ -0,0 +1,178 @@
+	mstore "github.com/mendersoftware/go-lib-micro/store/v2"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	mopts "go.mongodb.org/mongo-driver/mongo/options"
+)
+
+const (
+	findBatchSize = 100
+)
+
+type migration_2_0_0 struct {
+	ds     *DataStoreMongo
+	dbName string
+	ctx    context.Context
+}
+
+func (m *migration_2_0_0) Up(from migrate.Version) error {
+	logger := log.FromContext(m.ctx)
+	ctx := context.Background()
+
+	collectionsIndexes := map[string]struct {
+		Indexes []mongo.IndexModel
+	}{
+		DbUsersColl: {
+			Indexes: []mongo.IndexModel{
+				{
+					Keys: bson.D{{Key: DbUserEmail, Value: 1}},
+					Options: mopts.Index().
+						SetUnique(true).
+						SetName(DbUniqueEmailIndexName),
+				},
+			},
+		},
+		DbTokensColl: {
+			Indexes: []mongo.IndexModel{
+				{
+					Keys: bson.D{
+						{Key: DbTokenExpiresAt, Value: 1},
+					},
+					Options: mopts.Index().
+						SetExpireAfterSeconds(0).
+						SetName(DbTokenExpirationIndexName),
+				},
+				{
+					Keys: bson.D{
+						{Key: mstore.FieldTenantID, Value: 1},
+						{Key: DbTokenSubject, Value: 1},
+						{Key: DbTokenName, Value: 1},
+					},
+					Options: mopts.Index().
+						SetUnique(true).
+						SetPartialFilterExpression(
+							bson.M{
+								DbTokenName: bson.M{"$exists": true},
+							}).
+						SetName(DbTenantUniqueTokenNameIndexName),
+				},
+				{
+					Keys: bson.D{
+						{Key: mstore.FieldTenantID, Value: 1},
+						{Key: DbTokenSubject, Value: 1},
+					},
+					Options: mopts.Index().
+						SetName(DbTenantTokenSubjectIndexName),
+				},
+			},
+		},
+		DbSettingsColl: {
+			Indexes: []mongo.IndexModel{},
+		},
+	}
+
+	// for each collection in main useradm database
+	if m.dbName == DbName {
+		for collection, indexModel := range collectionsIndexes {
+			coll := m.ds.client.Database(m.dbName).Collection(collection)
+			// drop all the existing indexes, ignoring the errors
+			_, _ = coll.Indexes().DropAll(ctx)
```

Why is there no error handling here?

---

I've changed that in a separate commit

---

Whelp... Looks like it fails if the collection doesn't exist. I think it's safe to ignore this error code specifically.

---

commit removed